app/javascript/components/bookmarks/BookmarkManager.jsx

```diff
@@ -246,6 +246,31 @@ export default function BookmarkManager({bookmarks, studyAccession, clearExplore
   const [serverBookmarksLoaded, setServerBookmarksLoaded] = useState(false)
   const [showBookmarksModal, setShowBookmarksModal] = useState(false)
 
+  /** load bookmarks from server on initialization */
+  async function eagerLoadBookmarks() {
+    setHasEagerLoaded(true) // prevent thundering herd
```

Comment:
Could you elaborate a bit on how this would thundering-herd?

Reply:
This was a side effect of doing the loading via the `useEffect` with the empty dependency array - this lead to the async API call to the `/bookmarks` endpoint firing multiple times until the response came back and the state was set.  Having a preemptive short-circuit here cut that down to one and only one.  There might be a cleaner way of doing this, but I found this worked well enough.

Reply:
Thanks!